test/fixtures/non-string-argument/expected.js

```diff
@@ -0,0 +1,21 @@
+new Promise(resolve => {
+  resolve(require(new String({ 'answer': 42 })));
```

Comment:
huh, indeed http://www.ecma-international.org/ecma-262/7.0/index.html#sec-string-constructor-string-value has a separate step in `String()` to avoid throwing on a symbol.

In that case, it shouldn't call `new String`, it should use a package that provides the proper semantics - `is-symbol` to detect symbols, or just import `ToString` from `es-abstract`. In either case it's an added dependency to people's resulting code :-/

I think better would be `foo == null ? String(foo) : String(Object(foo))` which throws on Symbols and handles null/undefined - `new String` should be avoided.